test_fakeredis.py

```diff
@@ -3078,6 +3078,36 @@ def _listen(pubsub, q):
         received = q.get()
         self.assertEqual(received['data'], msg)
 
+    @attr('slow')
+    def test_pubsub_run_in_thread(self):
+        q = Queue()
+
+        def _queue_msg(msg, q):
+            q.put(msg)
+
+        pubsub = self.redis.pubsub()
+        pubsub.subscribe(**{"channel": lambda e: _queue_msg(e, q)})
+        pubsub_thread = pubsub.run_in_thread()
+
+        msg = b"Hello World"
+        self.redis.publish("channel", msg)
+
+        sleep(1)
+        retrieved = q.get()
+        self.assertEqual(retrieved["data"], msg)
+
+        pubsub_thread.stop()
+        pubsub_thread.join()
+        self.assertTrue(not pubsub_thread.is_alive())
+
+        pubsub.subscribe(**{"channel": None})
+        with self.assertRaises(redis.exceptions.PubSubError):
+            pubsub_thread = pubsub.run_in_thread()
+
+        pubsub.psubscribe(**{"channel": None})
+        with self.assertRaises(redis.exceptions.PubSubError):
+            pubsub_thread = pubsub.run_in_thread()
```

Comment:
Not sure this test will be effective, since you'll still have the normal `subscribe` to the channel so it will be bailing out due to that rather than the `psubscribe` without handler.

Reply:
unsubscribed first.